packages/malloy-render/src/stories/util.ts

```diff
@@ -44,10 +45,63 @@ export async function runQuery({
   view,
   connection,
 }: QueryOptions) {
-  const model = await loadModel({script, connection});
-  const runner = model.loadQuery(`run: ${source} -> ${view}`);
-  const result = await runner.run();
-  return result;
+  const urlReader: URLReader = {
+    readURL: async _url => {
+      return script;
+    },
+  };
+  const query: Malloy.Query = {
+    definition: {
+      kind: 'arrow',
+      source_reference: {name: source},
+      view: {
+        kind: 'view_reference',
+        name: view,
+      },
+    },
+  };
+  const lookupInfoConnection: API.LookupConnection<API.InfoConnection> = {
+    async lookupConnection(_name: string) {
+      return wrappedConnection;
+    },
+  };
+  const conn = await connection;
+  const wrappedConnection = API.util.wrapOldStyleConnection(conn);
+  const preparedResult = await API.asynchronous.compileQuery(
+    {
+      model_url: 'file:///script.malloy',
+      query,
+    },
+    {
+      urls: urlReader,
+      connections: lookupInfoConnection,
+    }
+  );
+  const lookupConnection: API.LookupConnection<API.Connection> = {
+    async lookupConnection(_name: string) {
+      return {
+        ...wrappedConnection,
+        runSQL: async (sql: string) => {
+          const result = await conn.runSQL(sql);
+          return API.util.mapData(result.rows, preparedResult.result!.schema);
+        },
+      };
+    },
+  };
+  const result = await API.asynchronous.runQuery(
+    {
+      model_url: 'file:///script.malloy',
+      query,
+    },
+    {
+      urls: urlReader,
+      connections: lookupConnection,
+    }
+  );
+  if (result.logs?.some(l => l.severity === 'error')) {
+    throw new Error(JSON.stringify(result.logs));
+  }
+  return result.result!;
```

Comment:
See if there's a better way to wrap this. Maybe just use the old thing to get an (old) `Result` and use `API.util.wrapResult`

Reply:
Currently the `mapData` function depends on the schema of the data. Which means it's not really possible to just write a single function to wrap an old-style connection into a new-style connection.